src/API/Reports/Categories/DataStore.php

```diff
@@ -70,92 +71,79 @@ class DataStore extends ReportsDataStore implements DataStoreInterface {
 		'orders_count'   => 'COUNT(DISTINCT order_id) as orders_count',
 		'products_count' => 'COUNT(DISTINCT product_id) as products_count',
 	);
+	/**
+	 * Data store context used to pass to filters.
+	 *
+	 * @var string
+	 */
+	protected static $context = 'categories';
 
 	/**
 	 * Constructor
 	 */
-	public function __construct() {
-		global $wpdb;
-		$table_name = $wpdb->prefix . self::TABLE_NAME;
-		// Avoid ambigious column order_id in SQL query.
-		$this->report_columns['products_count'] = str_replace( 'product_id', $table_name . '.product_id', $this->report_columns['products_count'] );
-		$this->report_columns['orders_count']   = str_replace( 'order_id', $table_name . '.order_id', $this->report_columns['orders_count'] );
+	protected function assign_report_columns() {
+		$this->report_columns['products_count'] = $this->prepend_table_name( $this->report_columns['products_count'], 'product_id' );
```

Comment:
I added the `assign_report_columns` functions to eliminate the class constructors in the implementation data stores. If we switch back to having constructors then all of the constructors have `parent::__construct();` or `$this->set_db_table_name();`.

Reply:
I think your'e right and that's a good reason to avoid the constructor method.  Would it be possible to move the entirety of `$report_columns` into `assign_report_columns()`?

What I was hoping to accomplish by this is to dry up those report columns a bit so we don't have to look in 2 places every time we make an update.  E.g., updating `coupon_id` in the above example would need to be modified in `$report_columns` and `assign_report_columns()` and a bit more prone to error during development.